types/transformer/artifacts/java.go

```diff
@@ -62,28 +62,15 @@ type JavaPackaging string
 
 // JarArtifactConfig defines a JarArtifactConfig struct
 type JarArtifactConfig struct {
-	DeploymentFile                    string            `yaml:"deploymentFile"`
-	JavaVersion                       string            `yaml:"javaVersion"`
-	BuildContainerName                string            `yaml:"buildContainerName"`
-	DeploymentFileDirInBuildContainer string            `yaml:"deploymentFileDirInBuildContainer"`
-	EnvVariables                      map[string]string `yaml:"envVariables"`
-	Port                              int32             `yaml:"port"`
+	Port               int32             `yaml:"port"`
+	JavaVersion        string            `yaml:"javaVersion"`
+	BuildContainerName string            `yaml:"buildContainerName"`
+	DeploymentFilePath string            `yaml:"deploymentFilePath"`
+	EnvVariables       map[string]string `yaml:"envVariables"`
 }
 
 // WarArtifactConfig defines a WarArtifactConfig struct
-type WarArtifactConfig struct {
-	DeploymentFile                    string            `yaml:"deploymentFile"`
-	JavaVersion                       string            `yaml:"javaVersion"`
-	BuildContainerName                string            `yaml:"buildContainerName"`
-	DeploymentFileDirInBuildContainer string            `yaml:"deploymentFileDirInBuildContainer"`
-	EnvVariables                      map[string]string `yaml:"envVariables"`
-}
+type WarArtifactConfig JarArtifactConfig
```

Comment:
Port does not have any meaning in War. If we want to reuse, we can see if promoted types work, and have port alone listed directly in Jar.

Reply:
see below

Reply:
replied below.